variadic macros error on Linux32 and OSX

on Linux32:

```console
$ ./creator_linux.sh 
bison -d lslmini.y
g++ -g -Wall -std=c++98 -pedantic-errors -fno-omit-frame-pointer -ffloat-store   -DVERSION='"1.0.9"' -DBUILD_DATE='"2017-12-08"' -o"lslmini.tab.o" -c lslmini.tab.c
In file included from ast.hh:5:0,
                 from lslmini.hh:29,
                 from lslmini.y:2:
logger.hh:144:19: error: ISO C++ does not permit named variadic macros [-Wvariadic-macros]
 #define DEBUG(args...)
                   ^
Makefile:125: recipe for target 'lslmini.tab.o' failed
make: *** [lslmini.tab.o] Error 1
```

on OSX:

```console
$ ./creator_osx.sh 
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100 49661  100 49661    0     0  23555      0  0:00:02  0:00:02 --:--:-- 24694
bison -d lslmini.y
g++ -g -Wall -std=c++98 -pedantic-errors -fno-omit-frame-pointer -ffloat-store -arch i386   -DVERSION='"1.0.9"' -DBUILD_DATE='"2017-12-08"' -o"lslmini.tab.o" -c lslmini.tab.c
clang: warning: treating 'c' input as 'c++' when in C++ mode, this behavior is deprecated [-Wdeprecated]
clang: warning: optimization flag '-ffloat-store' is not supported [-Wignored-optimization-argument]
In file included from lslmini.y:2:
In file included from ./lslmini.hh:29:
In file included from ./ast.hh:5:
./logger.hh:144:19: error: named variadic macros are a GNU extension
      [-Werror,-Wvariadic-macros]
#define DEBUG(args...)
                  ^
In file included from lslmini.y:2:
In file included from ./lslmini.hh:28:
./symtab.hh:62:26: warning: private field 'cur_references' is not used
      [-Wunused-private-field]
    int                  cur_references;        // how many times the cu...
                         ^
1 warning and 1 error generated.
make: *** [lslmini.tab.o] Error 1
```

--- a/logger.hh
+++ b/logger.hh
@@ -140,11 +140,7 @@ enum AssertType {
 #ifdef DEBUG_LEVEL
 #define DEBUG LOG
 #else /* not DEBUG_LEVEL */
-#ifdef __GNUC__
-#define DEBUG(args...)
-#else /* not __GNUC__ */
-#define DEBUG(...)
-#endif /* not __GNUC__ */
+static inline void DEBUG(...) { }
 #endif /* not DEBUG_LEVEL */
 #endif /* not _MSC_VER */
 

